app/commands/git/sync_concept_exercise.rb

```diff
@@ -17,13 +17,29 @@ def call
         git_sha: head_git_exercise.commit.oid,
         synced_to_git_sha: head_git_exercise.commit.oid,
         taught_concepts: find_concepts(exercise_config[:concepts]),
-        prerequisites: find_concepts(exercise_config[:prerequisites])
+        prerequisites: find_concepts(exercise_config[:prerequisites]),
+        authors: authors
       )
+
+      update_reputation!
     end
 
     private
     attr_reader :exercise
 
+    def update_reputation!
+      exercise.authorships.each do |authorship|
+        User::ReputationAcquisition.find_or_create_by!(
```

Comment:
I used `find_or_create_by!` instead of `create_or_find_by!` as there is no unique key on any of the keys, nor do I think there should be a unique key as we might have objects on which we want to bestow multiple reputation acquisitions I think.

Reply:
One thing we also need to decide is whether we want to allow _removal_ of reputation acquisitions, if someone is removed as an author (not very likely).

Reply:
I was thinking this when I wrote my thoughts in Slack to you. Let's presume that you can't remove authors for now. And then we'll worry about this a bit further down the line.